Add Omics functions

maggiecbms opened the pull request:
This PR includes changes to add Omics support.

golharam committed:
Adding AWS Omics support

golharam committed:
Remove hard-coded role arn

golharam committed:
Remove hard-coded output path

golharam committed:
Add boto3 to requirements.txt

golharam committed:
Get role arn from connect() parameters

golharam committed:
Add stage_output_files

golharam committed:
Update get_project

golharam committed:
Revert get_project

golharam committed:
Add notes for copy_folder

golharam committed:
Add trailing slash only if its missing

golharam committed:
Add output_bucket

golharam committed:
Add output_bucket

golharam committed:
Implement throttling to 1 TPS on Omics

golharam committed:
Implement throttling to 1 TPS on Omics

golharam committed:
Merge branch 'main' into add_omics

golharam committed:
Merge branch 'main' into add_omics

golharam committed:
Merge branch 'main' into add_omics

golharam committed:
Adding unit tests for AWS Omics

golharam committed:
Add missing implementations for functions

golharam committed:
Add comment

maggiecbms committed:
Merge branch 'main' into add_omics_functions

maggiecbms committed:
make omics consistent with base

maggiecbms committed:
add function to support launcher;

maggiecbms committed:
add s3 client to init

maggiecbms committed:
Merge branch 'main' into add_omics_functions

maggiecbms committed:
update upload_file function name

maggiecbms committed:
add omics function

maggiecbms committed:
remove space in output url

golharam committed:
Add TODO comments

maggiecbms committed:
update storagetype for omics

golharam committed:
Merge branch 'main' into add_omics_functions

golharam committed:
Create a run group for omics when calling create_project

golharam committed:
Return the rungroupid

golharam committed:
Add missing get_project_cost

golharam committed:
Add stubs to missing methods

golharam committed:
Revert run_group_id -> projectid

golharam committed:
Remove hardcoded values from omics_platform.py

golharam committed:
fix output path in omics_platform.py

golharam committed:
Force failure if output_bucket and role_arn are not provided

golharam commented:
I think this PR is almost ready to go.  The one thing I don't like is that launchers can be run with the '--project_name' field, but when submitting tasks, a 'project_id' aka run group id is expected so there is still a little discrepancy that needs to be addressed.

golharam commented:
The last thing I notice is there are references to 'Output', and 'output'.  Additionally files are uploaded to /outputs/jobid/inputs/....   
Its a little confusing.

maggiecbms committed:
add get_project_by_name for omics

maggiecbms commented:
> The last thing I notice is there are references to 'Output', and 'output'. Additionally files are uploaded to /outputs/jobid/inputs/.... It's a little confusing. For instance, upload_file, the target_path is "Outputs/[run_group_id]/launcher_inputs/[filename]"
> 
> Maybe we shouldn't use "Outputs" as a folder name as 'output' is used by omics for the workflow outputs.

Currently output files of a run goes into s3://{Bucket}/Outputs/{RunGroupID}/{WorkflowID}/{TaskName}/{RunID}/ , and the upload_file() uploads files into s3://{Bucket}/Outputs/{RunGroupID}/launcher_inputs/ .

I can change the output folder to s3://{Bucket}/**Project**/{RunGroupID}/{WorkflowID}/{TaskName}/{RunID}/ and upload file to s3://{Bucket}/**Project**/{RunGroupID}/launcher_inputs/ .

maggiecbms committed:
allow not finding project by name

maggiecbms committed:
update get task output

golharam commented on src/cwl_platform/omics_platform.py:
```diff
@@ -0,0 +1,358 @@
+            outputs = [c["location"] for c in all_outputs]
+            return outputs
+        else:
+            return all_outputs["location"]
```
Check for the existence of 'location' as a key before using or else this could result in a KeyError

golharam commented on src/cwl_platform/omics_platform.py:
```diff
@@ -0,0 +1,358 @@
+
+        return omics response for start_run.
+        '''
+        base_output_path = f"s3://{self.output_bucket}/Project/"
```
I don't like that we could have two different output locations.  We should be consistent about this.  ProjectId is actually runGroupId I believe.  Can we update this key to be runGroupId so its meaning is clear?

golharam commented on src/cwl_platform/omics_platform.py:
```diff
@@ -0,0 +1,358 @@
+            run_group_id = response['items'][0]['id']
+
+            project = {
+                'ProjectId': run_group_id
```
ProjectId here should be 'runGroupId'.

golharam commented on src/cwl_platform/omics_platform.py:
```diff
@@ -0,0 +1,358 @@
+            logger.error('Could not find project with name: %s', project_name)
+            return {}
+
+    def get_project_by_id(self, project_id):
```
project_id actually means run group id.

golharam reviewed (approved):
This looks good:
1.  Change references to ProjectId to runGroupId to remove any confusion of terms
2. Ask Claude 3.7 to implement unit tests

After that, we are good.

maggiecbms committed:
update omics formats

maggiecbms committed:
add check for locations

maggiecbms committed:
add omics unit test

maggiecbms committed:
fix omics unit test

maggiecbms commented:
> This looks good:
> 
> 1. Change references to ProjectId to runGroupId to remove any confusion of terms
> 2. Ask Claude 3.7 to implement unit tests
> 
> After that, we are good.

I have replaced the ProjectId to RunGroupId in omics_platform.py.
Unit test is also added with help of Claude.

maggiecbms committed:
add cache

maggiecbms committed:
Merge remote-tracking branch 'origin/main' into add_omics_functions

maggiecbms committed:
update omics wes api

maggiecbms committed:
get api username password from environment

golharam commented:
This PR is superceded by #83.
